src/sotauptaneclient.h

```diff
@@ -39,6 +41,8 @@ class SotaUptaneClient {
   const Config &config;
   event::Channel *events_channel;
   Uptane::Repository &uptane_repo;
+  const boost::shared_ptr<INvStorage> &storage;
```

Comment:
Would this be better as a bare `boost::shared_ptr<INvStorage> storage`, rather than a reference to a shared pointer?

Reply:
Yeah, if I keep it const, that should be fine. Reference to a pointer is not good style!

Reply:
@cajun-rat I pointed out a similar issue on Serhiy's PR (https://github.com/advancedtelematic/aktualizr/pull/485#discussion_r159892094) and he shared this link: https://stackoverflow.com/questions/3310737/shared-ptr-by-reference-or-by-value

I think there's room to argue about non-const references to a shared pointer, but if it is const, it would seem that passing as a reference is the right way to go after all.

Reply:
@patriotyk @patrickvacek  Interesting!  Turns out I was wrong about ref to shared_ptr.

Reply:
Ah, but hold on--this is a member variable in an object, where we want to use shared_ptr to manage the object lifecycle.  Those links are discussing function parameters, where I can see the benefit.

Reply:
So my impression is that if we are passing a non-const shared_ptr to a class constructor, it should probably not be passed by reference. But if it is a const shared_ptr, it probably is better a reference. That only affects SotaUptaneClient in this PR; the other two cases are fine as they are (non-const, non-reference shared_ptrs).